asdf/commands/edit.py

```diff
@@ -0,0 +1,529 @@
+        print(f"Error: No file '{fname}' exists.")
+        return False
+
+    # Simply validates the path existence
+    if wanted_ext is None:  
+        return True
+    
+    # Make sure the extension is one desired.
+    base, ext = os.path.splitext(fname)
+    if ext not in wanted_ext:
+        return False
+
+    return True
+
+
+def is_validate_asdf_path ( fname ) :
+    """ Validates fname path exists and has extension '.asdf'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    """
+    ext = ['.asdf']
+    if is_validate_path_and_ext(fname,ext) : 
+        return True
+    print(f"Error: '{fname}' should have extension '{ext[0]}'")
+    return False
+
+
+def is_validate_yaml_path ( fname ) :
+    """ Validates fname path exists and has extension '.yaml'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    """
+    ext = ['.yaml']
+    if is_validate_path_and_ext(fname,ext) : 
+        return True
+    print(f"Error: '{fname}' should have extension '{ext[0]}'")
+    return False
+
+
+def validate_asdf_file ( fd ) :
+    """ Makes sure the header line is the expected one, as well
+    as getting the optional comment line.
+
+    Parameters
+    ----------
+    fd : GenericFile
+    """
+    #global asdf_format_version 
+    #global asdf_standard_version 
+    ASDF_ID = b'#ASDF'
+
+    header_line = fd.read_until(b'\r?\n', 2, "newline", include=True)
+    if ASDF_ID!=header_line[:len(ASDF_ID)] :
+        # Maybe raise exception
+        print("Invalid ASDF ID")
+        sys.exit(1)
+    
+    #asdf_format_version = _parse_asdf_header_line(header_line)
+    # Maybe validate ASDF format version
+    comment_section = fd.read_until( b'(%YAML)|(' + constants.BLOCK_MAGIC + b')', 
+                                     5, 
+                                     "start of content", 
+                                     include=False, 
+                                     exception=False)
+    # Maybe do the following for more validate.  But maybe not.
+    #comments = _parse_asdf_comment_section(comment_section)
+    #asdf_standard_version = _get_asdf_version_in_comments(comments)
+
+    return header_line + comment_section 
+    
+def open_and_validate_asdf ( fname ) :
+    """ Open and validate the ASDF file, as well as read in all the YAML
+    that will be outputted to a YAML file.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    """
+    fullpath = os.path.abspath(fname)
+    fd = generic_io.get_file(fullpath, mode="r")
+
+    # Read the ASDF header and optional comments section
+    header_and_comment = validate_asdf_file(fd)
+
+    return fd, header_and_comment # Return GenericFile and ASDF header bytes.
+    
+def read_and_validate_yaml ( fd, fname ) :
+    """ Get the YAML text from an ASDF formatted file.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    fd : GenericFile for fname.
+    """
+    YAML_TOKEN = b'%YAML'
```

Comment:
I mentioned this down below, but I think we should omit this from the temp file so that the user has fewer options for making a mistake.

Reply:
We discussed this over email and agreed to keep the entire YAML document editable.